### ISSUE TITLE
Model can't find files while training

### ISSUE THREAD

**Issue by AlekseiPol** [OlafenwaMoses/ImageAI#333]
Greetings.
I try to train detector using next code
`import os`
`os.chdir("/content/gdrive/My Drive/python_projects/kuzushiji reconginition/")`

`from imageai.Detection.Custom import DetectionModelTrainer`

`objects_to_detect = ['U+306F', 'U+304C']`

`trainer = DetectionModelTrainer()`
`trainer.setModelTypeAsYOLOv3()`
`trainer.setDataDirectory(data_directory="detection_input_2/")`
`trainer.setTrainConfig(object_names_array=objects_to_detect, batch_size=4, num_experiments=10)`
`trainer.trainModel()`

I get the following error during 1st epoch:

Generating anchor boxes for training images and annotation...
Average IOU for 9 anchors: 0.86
Anchor Boxes generated.
Detection configuration saved in  detection_input_2/json/detection_config.json
Training on: 	['U+304C', 'U+306F']
Training with Batch Size:  4
Number of Experiments:  10
Pre-trained Model not provided. Transfer learning not in use.
Training will start with 3 warmup experiments
/usr/local/lib/python3.6/dist-packages/keras/callbacks.py:1335: UserWarning: `epsilon` argument is deprecated and will be removed, use `min_delta` instead.
  warnings.warn('`epsilon` argument is deprecated and '
Epoch 1/13
Cannot find  detection_input_2\validation/images/200003076_00043_2.jpg
Cannot find Cannot find Cannot find Cannot find   detection_input_2\train/images/umgy005-001.jpgdetection_input_2\validation/images/200021851_00022_2.jpg

Cannot find   detection_input_2\validation/images/200003076_00154_1.jpgdetection_input_2\train/images/100249371_00007_2.jpg

Cannot find Cannot find    detection_input_2\validation/images/200014740-00065_2.jpg
detection_input_2\train/images/hnsd011-002.jpg
detection_input_2\train/images/umgy001-007.jpg
Cannot find Cannot find  Cannot find Cannot find  detection_input_2\validation/images/200003076_00134_1.jpg detection_input_2\train/images/200021660-00049_2.jpg
 detection_input_2\train/images/hnsd010-015.jpgdetection_input_2\validation/images/hnsd008-026.jpg

Cannot find Cannot find   detection_input_2\validation/images/hnsd005-013.jpg
detection_input_2\train/images/200021712-00075_2.jpg
Cannot find  detection_input_2\train/images/brsk005-021.jpgCannot find  detection_input_2\validation/images/200005598_00033_1.jpg

The problem seems to be in \ instead of / because
`print(os.path.exists("detection_input_2\validation/images/200003076_00154_1.jpg"))`
`print(os.path.exists("detection_input_2/validation/images/200003076_00154_1.jpg"))`
False
True

How to fix such problem? I've tried using data_directory="detection_input_2/" and data_directory="detection_input_2". Encountered such problem in google colab and windows 10.

**Comment by rola93** [comment]
I can't try right now. But it looks like an issue with using / and \ at the same time. As far as I can see, the problem is con custom/__init__.py

Look at the following lines:
```python
self.__train_images_folder = os.path.join(data_directory, "train/images/")
        self.__train_annotations_folder = os.path.join(data_directory, "train/annotations/")
        self.__validation_images_folder = os.path.join(data_directory, "validation/images/")
        self.__validation_annotations_folder = os.path.join(data_directory, "validation/annotations/")
```

Slashes are hard coded! They must be changed to `is.path.join` operations, as it is done later in the code. 

I'll write a or to fix it, but would be nice you to double check it by just editing the source code of your installation

**Comment by AlekseiPol** [comment]
Seems to be problem also in create_training_instances
I've removed hardcoded slashes but train_ints returned from create_training_instances still have wrong paths:
'filename': 'detection_input_2\\train/images/100241706_00005_1.jpg', 'width': 200, 'height': 200}

**Comment by AlekseiPol** [comment]
Your fix is ok, parse_voc_annotation loaded labels from cache which I forgot to clear.
Btw paths in parse_voc_annotation are constructed this way: img_dir + elem.text, ann_dir + ann. 
So the one should add / in the end of folder paths or use os.path.join there too.
Thanks for your assistance.

**Comment by rola93** [comment]
hi @AlekseiPol 

Would be great to hear your thoughts on #336 which should solve this issue

thanks

**Comment by rola93** [comment]
shouldn't this be closed? @AlekseiPol ?

**Comment by OlafenwaMoses** [comment]
Since the path issue has been fixed in #336 and released in the latest version `v2.1.5` , I will be closing this issue.

**Comment by abhisingh007224** [comment]
I don't think it is fixed still facing the same error

**Comment by thangaraj535** [comment]
same error even with upgraded imageai
